test/e2e/common/node/framework/cgroups/cgroups.go

```diff
@@ -0,0 +1,421 @@
+	podCgPath, stderr, err := e2epod.ExecCommandInContainerWithFullOutput(f, pod.Name, pod.Spec.Containers[0].Name, []string{"/bin/sh", "-c", cmd}...)
+	if podCgPath == "" {
+		// This command may hit 'No such file or directory' for another cgroup if another test running in parallel has deleted a pod.
+		// We ignore errors if podCgPath is found.
+		if err != nil || len(stderr) > 0 {
+			return "", fmt.Errorf("encountered error while running command: %q, \nerr: %w \nstdErr: %q", cmd, err, stderr)
+		}
+		return "", fmt.Errorf("pod cgroup dirctory not found by command: %q", cmd)
+	}
+	return podCgPath, nil
+}
+
+func getCgroupMemLimitPath(cgPath string, podOnCgroupv2 bool) string {
+	if podOnCgroupv2 {
+		return fmt.Sprintf("%s/%s", cgPath, cgroupv2MemLimitFile)
+	} else {
+		return fmt.Sprintf("%s/%s", cgPath, cgroupMemLimitFile)
+	}
+}
+
+func getCgroupCPULimitPath(cgPath string, podOnCgroupv2 bool) string {
+	if podOnCgroupv2 {
+		return fmt.Sprintf("%s/%s", cgPath, cgroupv2CPULimitFile)
+	} else {
+		return fmt.Sprintf("%s/%s", cgPath, cgroupCPUQuotaFile)
+	}
+}
+
+func getCgroupCPURequestPath(cgPath string, podOnCgroupv2 bool) string {
+	if podOnCgroupv2 {
+		return fmt.Sprintf("%s/%s", cgPath, cgroupv2CPUWeightFile)
+	} else {
+		return fmt.Sprintf("%s/%s", cgPath, cgroupCPUSharesFile)
+	}
+}
+
+// TODO: Remove the rounded cpu limit values when https://github.com/opencontainers/runc/issues/4622
+// is fixed.
+func getCPULimitCgroupExpectations(cpuLimit *resource.Quantity, podOnCgroupV2 bool) []string {
+	var expectedCPULimits []string
+	milliCPULimit := cpuLimit.MilliValue()
+
+	cpuQuota := kubecm.MilliCPUToQuota(milliCPULimit, kubecm.QuotaPeriod)
+	if cpuLimit.IsZero() {
+		cpuQuota = -1
+	}
+	expectedCPULimits = append(expectedCPULimits, getExpectedCPULimitFromCPUQuota(cpuQuota, podOnCgroupV2))
+
+	if milliCPULimit%10 != 0 && cpuQuota != -1 {
+		roundedCPULimit := (milliCPULimit/10 + 1) * 10
+		cpuQuotaRounded := kubecm.MilliCPUToQuota(roundedCPULimit, kubecm.QuotaPeriod)
+		expectedCPULimits = append(expectedCPULimits, getExpectedCPULimitFromCPUQuota(cpuQuotaRounded, podOnCgroupV2))
+	}
+
+	return expectedCPULimits
+}
+
+func getExpectedCPULimitFromCPUQuota(cpuQuota int64, podOnCgroupV2 bool) string {
+	expectedCPULimitString := strconv.FormatInt(cpuQuota, 10)
+	if podOnCgroupV2 {
+		if expectedCPULimitString == "-1" {
+			expectedCPULimitString = "max"
+		}
+		expectedCPULimitString = fmt.Sprintf("%s %d", expectedCPULimitString, kubecm.QuotaPeriod)
+	}
+	return expectedCPULimitString
+}
+
+func getExpectedCPUShares(rr *v1.ResourceRequirements, podOnCgroupv2 bool) int64 {
+	cpuRequest := rr.Requests.Cpu()
+	cpuLimit := rr.Limits.Cpu()
+	var shares int64
+	if cpuRequest.IsZero() && !cpuLimit.IsZero() {
+		shares = int64(kubecm.MilliCPUToShares(cpuLimit.MilliValue()))
+	} else {
+		shares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
+	}
+	if podOnCgroupv2 {
+		return 1 + ((shares-2)*9999)/262142
```

Comment:
nit: we have this formula in a bunch of places, and should just have a shared function. Just add a TODO for now.

Reply:
I added a TODO.